scripts/area_plot.py

```diff
@@ -0,0 +1,613 @@
+                    required=True,
+                    )
+parser_one.add_argument('-f', '--figure',
+                    help=('The location for the final figure'),
+                    required=True,
+                    )
+parser_one.add_argument('-s', '--samples', 
+                    help=('A text file with the list of samples to be include' 
+                         ' (one per line). If no list is provided, then data '
+                         'from all columns in the table (except the one '
+                         'specifying taxonomy) will be used.'),
+                    )
+parser_one.add_argument('--mode', choices=(["metaphlan", "kracken", "marker"]),
+                    help=('The software generating the table to make parsing'
+                          'easier. Optens are kracken, metaphlan, marker '
+                          '(i.e. CTMR amplicon) and other for all other '
+                          'processing types.'),
+                    default='kracken'
+                    )
+parser_one.add_argument('-l', '--level',
+                    help=('The taxonomic level (as an integer) to plot the '
+                          'data.'),
+                    default=3,
+                    type=int,
+                    )
+parser_one.add_argument('--abund-thresh',
+                    help=("the minimum abundance required to display a group."),
+                    default=0.01,
+                    type=float,
+                    )
+parser_one.add_argument('--group-thresh',
+                    help=("The maximum number of groups to be displayed in "
+                        "the graph."),
+                    default=8,
+                    type=int,
+                    )
+parser_one.add_argument('-c', '--colormap',
+                    help=("The qualitative colormap to use to generate your "
+                          "plot. Refer to colorbrewer for options. If a "
+                          "selected colormap exceeds the number of groups "
+                          "(`--group-thresh`) possible, it will default to "
+                          "Set3."),
+                    default='Set3',
+                    )
+parser_one.add_argument('--sub-level',
+                        help=('The second level to use if doing a joint plot'),
+                        type=int,
+                        )
+parser_one.add_argument('--sub-abund-thresh',
+                        help=("the minimum abundance required to display a sub group"),
+                        default=0.05,
+                        type=float,
+                        )
+parser_one.add_argument('--sub-group-thresh',
+                        help=("the maximum number of sub groups allowed in "
+                              "a joint level plot."),
+                        default=5,
+                        type=float,
+                        )
+
+if __name__ == '__main__':
+    args = parser_one.parse_args()
+
+    if args.mode.lower() == 'metaphlan':
```

Comment:
You won't need to call `str.lower()` on the parsed args, because ArgumentParser constrains them to the choices you have defined--they _have to_ be lowercase already, since that's how you defined the choices.